csharp-tree-sitter.el

```diff
@@ -274,30 +277,51 @@
     (modify-syntax-entry ?@ "_" table)
     table))
 
-;;;###autoload
-(define-derived-mode csharp-tree-sitter-mode prog-mode "C#"
-  "Major mode for editing Csharp code.
+(defvar-local csharp-tree-sitter--syntax-table nil)
+(defvar-local csharp-tree-sitter--mode-map nil)
+(defvar-local csharp-tree-sitter--indent-line-function nil)
 
-Key bindings:
-\\{csharp-tree-sitter-mode-map}"
-  :group 'csharp
-  :syntax-table csharp-tree-sitter-mode-syntax-table
+(defun csharp-tree-sitter--enable ()
+  "Enable tree-sitter support for csharp."
+  ;; NOTE: Record lost data before configure to `tree-sitter' support.
+  (setq csharp-tree-sitter--syntax-table csharp-mode-syntax-table
+        csharp-tree-sitter--mode-map csharp-mode-map
+        csharp-tree-sitter--indent-line-function indent-line-function)
```

Comment:
Record down all necessary variables before enabling tree-sitter support. This should respect to user's configuration.